src/server/api/routers/assignment.ts

```diff
@@ -0,0 +1,136 @@
+import { Prisma } from '@prisma/client';
+import { TRPCError } from "@trpc/server";
+import { z } from 'zod';
+import {
+  createTRPCRouter,
+  publicProcedure,
+  protectedProcedure,
+  adminProcedure,
+  mentorProcedure
+} from '~/server/api/trpc';
+
+export const assignmentRouter = createTRPCRouter({
+  getAssignment: mentorProcedure
+    .input(
+      z.object({
+        filterBy: z.string().optional(),
+        searchQuery: z.string().optional(),
+        currentPage: z.number(),
+        limitPerPage: z.number()
+      })
+    )
+    .query(async ({ ctx, input }) => {
+      return await ctx.prisma.assignment.findMany({
+        where: {
+          title: {
+            contains: input.searchQuery
+          }
+        },
+        skip: (input.currentPage - 1) * input.limitPerPage,
+        take: input.limitPerPage
+      });
+    }),
+
+  setAssignmentScore: mentorProcedure
+    .input(
+      z.object({
+        submissionId: z.string(),
```

Comment:
kalo input id tambah validation uuid
```suggestion
        submissionId: z.string().uuid(),
```